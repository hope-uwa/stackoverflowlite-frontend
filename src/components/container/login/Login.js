import React, { Component } from 'react';
import PropTypes from 'prop-types';


export default class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      loginDisplay: ''
    };
  }


  componentWillReceiveProps =(props) => {
    const { loginDisplay } = props;
    this.setState({ loginDisplay });
  }

  closeModal = () => {
    this.setState({
      email: '',
      password: '',
      loginDisplay: 'none'
    });
  }

  checkBoxChecked() {
    return true;
  }

  loginSubmit = async (event) => {
    event.preventDefault();
    const userData = { ...this.state };
    event.persist();
    event.target.innerHTML = 'LOADING......';
    await this.props.login(userData);
    if (this.props.loginInfo.token) {
      setTimeout(() => this.setState({ loginDisplay: 'none' }), 1500);
    } else {
      event.target.innerHTML = 'LOGIN';
    }
  }

  handleChange= (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  render() {
    const display = this.state.loginDisplay;
    const style = {
      display,
    };
    return (
      <div id="loginModal" className="modal" style={style}>
        <div className="modal-content">
          <div className="modal-head w100">
            <span className="pl1 modal-title ">Welcome Back!</span>
            <span id="s-close" onClick={this.closeModal} className="close">&times;</span>
          </div>

          <div className="modal__signup-form row">
            <form id="login-form" className="w100 mt2 mb2" >
              <div className="modal-form align-center mb2">
                <input type="email" required placeholder="Enter Email Address" id="loginEmail"
                  className="email" name="email" onChange={this.handleChange} />
                <input type="password" required placeholder="Enter Password" id="loginPassword"
                  className="email" name="password" onChange={this.handleChange} />
              </div>
              <div className="modal-footer  w-100 pt2 pb1">
                <button id="submit" type="submit"onClick={this.loginSubmit}
                  className="btn btn-shadow btn_login-signup w60">LOGIN</button>
              </div>
            </form>
          </div>
        </div>

      </div>

    );
  }
}
Login.propTypes = {
  loginDisplay: PropTypes.string,
  loginInfo: PropTypes.object,
  login: PropTypes.func
};
