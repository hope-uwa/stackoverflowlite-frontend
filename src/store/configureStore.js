import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const middleware = process.env.NODE_ENV !== 'production' ?
  [require('redux-immutable-state-invariant').default(), thunk] :
  [thunk];

function configureStore(initialState) {
  return createStore(
    rootReducer,
    initialState,
    applyMiddleware(...middleware),
    );
}
const store = configureStore();
window.store = store;
store.subscribe(()=> localStorage.setItem('user',JSON.stringify(store.getState().loginReducer)));
export default store;