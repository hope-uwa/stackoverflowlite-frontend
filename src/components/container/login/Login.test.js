/* eslint-disable no-undef */
import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import Login from './Login';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);


describe('<Login />', () => {
  it('matches the snapshoot', () => {
    const initialState = {};
    const store = mockStore(initialState);

    const wrapper = mount(<Provider store={store}>
      <Login/>
    </Provider>);
    
    expect(wrapper).toMatchSnapshot();
  });

  it('call function', () => {
    const mockSignupFn = jest.fn();
    const initialState = {};
    const store = mockStore(initialState);

    const wrapper = shallow(<Provider store={store}>
      <Login/>
    </Provider>);
    // console.log(wrapper.debug());
    // expect(wrapper.dive().instance().checkBoxChecked()).toEqual(true);
    // expect(wrapper.dive().instance().closeModal()).toReturn(mockSignupFn);
    // const form = wrapper.find('form');
    // form.simulate('submit');
    // wrapper.instance().loginSubmit() = mockSignupFn;
    // expect(wrapper.dive().instance().loginSubmit()).toContain()

    //     const spy = jest.spyOn(Login.prototype, 'componentWillReceiveProps');
    // const wrappers = mount(<Login />);
    // wrappers.instance().componentWillReceiveProps();
    // expect(spy).toHaveBeenCalled();
  });

  it('render login page correctly', () => {
    const wrapper = mount(<Login />);
    expect(wrapper.find('button').text()).toBe('LOGIN');
    expect(wrapper.find('.modal-title').text()).toBe('Welcome Back!');
    expect(wrapper.find('form input').length).toEqual(2);
  });

  it('should update states when data is passed', () => {
    const wrapper = mount(<Login />);
    const emailInput = wrapper.find('#loginEmail');
    emailInput.instance().value = 'uwahope007@gmail.com';
    emailInput.simulate('change');

    const passwordInput = wrapper.find('#loginPassword');
    passwordInput.instance().value = 'password123';
    passwordInput.simulate('change');

    expect(wrapper.state()).toMatchObject({
      email: 'uwahope007@gmail.com',
      password: 'password123',
      loginDisplay: '',
    });
    wrapper.unmount();
  });

  it('should submit inputs from the state', () => {
    const mockSignupFn = jest.fn();
    const wrapper = mount(<Login signup={mockSignupFn} />);

    const emailInput = wrapper.find('#loginEmail');
    emailInput.instance().value = 'uwahope007@gmail.com';
    emailInput.simulate('change');

    const passwordInput = wrapper.find('#loginPassword');
    passwordInput.instance().value = 'password123';
    passwordInput.simulate('change');

    const form = wrapper.find('form');
    form.simulate('submit');
    expect(mockSignupFn.mock.calls.length).toBe(0);
    wrapper.unmount();
  });
  it("loginSubmit called on submit", () => {
    const value = "Makers Academy";
    const loginSubmit = jest.fn();
    const wrapper = mount(<Login loginSubmit={loginSubmit}/>);

    const result = wrapper.find('#submit');
    result.simulate('click');
    expect(loginSubmit.called).toBe();
  });
// 
});
