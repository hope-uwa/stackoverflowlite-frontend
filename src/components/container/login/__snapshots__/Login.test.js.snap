// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches the snapshoot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Login>
    <div
      className="modal"
      id="loginModal"
      style={
        Object {
          "display": "",
        }
      }
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-head w100"
        >
          <span
            className="pl1 modal-title "
          >
            Welcome Back!
          </span>
          <span
            className="close"
            id="s-close"
            onClick={[Function]}
          >
            Ã—
          </span>
        </div>
        <div
          className="modal__signup-form row"
        >
          <form
            className="w100 mt2 mb2"
            id="login-form"
          >
            <div
              className="modal-form align-center mb2"
            >
              <input
                className="email"
                id="loginEmail"
                name="email"
                onChange={[Function]}
                placeholder="Enter Email Address"
                required={true}
                type="email"
              />
              <input
                className="email"
                id="loginPassword"
                name="password"
                onChange={[Function]}
                placeholder="Enter Password"
                required={true}
                type="password"
              />
            </div>
            <div
              className="modal-footer  w-100 pt2 pb1"
            >
              <button
                className="btn btn-shadow btn_login-signup w60"
                id="submit"
                onClick={[Function]}
                type="submit"
              >
                LOGIN
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </Login>
</Provider>
`;
